networkit/cpp/io/test/IOGTest.cpp

```diff
@@ -866,4 +866,24 @@ TEST_F(IOGTest, testNetworkitBinaryWiki) {
 		});
 	});
 }
+TEST_F(IOGTest, NetworkitBinaryAutoDetect) {
+
+	Graph G(10, true, false);
+	int64_t weight = -1;
+	for(count n = 0; n < G.numberOfNodes(); n++) {
+		if(n != G.numberOfNodes()-1)
+			G.addEdge(n, n+1, weight++);
+	}
```

Comment:
This tests negative weights, correct? My suggestion is to rename the test to `testNetworkitBinarySignedWeights`. Maybe we should also add a test for floating point weights?